zrml/neo-swaps/src/consts.rs

```diff
@@ -22,7 +22,7 @@ pub(crate) const EXP_NUMERICAL_LIMIT: u128 = 10;
 /// Numerical lower limit for ln arguments (fixed point number).
 pub(crate) const LN_NUMERICAL_LIMIT: u128 = BASE / 10;
 /// The maximum number of assets allowed in a pool.
-pub(crate) const MAX_ASSETS: u16 = 128;
+pub(crate) const MAX_ASSETS: u32 = 128;
```

Comment:
Why is that necessary? `u32` seems overkill and also deviates from `MaxCategories` type in PM.

Reply:
Took me a while to remember. It's because the generic size parameter of the `BoundedBTreeMap` expects a `Get<u32>` implementor. I guess I could write an adapter instead.

Reply:
The deviation actually makes sense to me. In combo, we're not combining markets, we're just combining their pools. So combo pools will, in general, have more assets than markets.

Reply:
Potentially more than 65535 assets? Sounds like a lot of effort, curious to see the clever shortcuts to do the math that you'll find :)

Reply:
:) Anything over 1,000 is pushing it. When I said the deviation makes sense I meant it makes sense to allow a different value rather than a different type. The `u32` is really only there to satisfy the trait bounds of `Bounded*`.